crates/uv-resolver/src/lock/target.rs

```diff
@@ -211,9 +214,10 @@ impl<'env> InstallTarget<'env> {
                 })
                 .flatten()
             {
-                if !dep.complexified_marker.evaluate(marker_env, &[]) {
+                if !dep.complexified_marker.evaluate_no_extras(marker_env) {
                     continue;
                 }
+                activated_groups.push((dist.id.name.clone(), group.clone()));
```

Comment:
My initial instinct is... should the group and extras not somehow be pushed onto `queue`? (I guess they're "indexed" by package name? It just seems vaguely off for them to be tracked as "global" state in the algorithm like this.)

Reply:
I am pretty sure you need them to be global. A conflict marker might be something like `extra == 'foo' or extra != 'bar'`, for example. So you really do need to know the full set of activated extras/groups.

I _think_ part of the confusion here is that while conflict markers are using `extra`, and conflicts can be declared on extras, these don't really resemble anything like what the PEP 508 `extra` stands for. We're basically just abusing it to encode the conflict marker semantics into PEP 508 markers. So the standard practice for how we previously used markers for extras doesn't really apply to conflict markers AIUI.